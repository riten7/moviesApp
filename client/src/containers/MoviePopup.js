import React, { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { showMoviePopup } from '../actions/actionCreators';
import { BASE_URL } from '../actions/Constant';
import { Modal, Form, Input, Button, Select, DatePicker } from 'antd';
import moment from 'moment';
import { useHistory } from 'react-router-dom';

const layout = { labelCol: { span: 6, }, wrapperCol: {span: 16, },};
const tailLayout = { wrapperCol: { offset: 16 },};

const MoviePopup = (props) => { 
  const [form] = Form.useForm();
  const { Option } = Select;
  const { TextArea } = Input;
  const history = useHistory();
  const dispatch = useDispatch();
  const moviePopupShown = useSelector(state => state.popup);

  const closePopup = () => (dispatch(showMoviePopup(false)));

  const getBaseUrl = () => {
    return false ? BASE_URL + "/movie/" + props.movie.id : BASE_URL + "/movie";
  }


  // const addMovieToDb = useCallback((data) => {
  //   fetch(getBaseUrl(), {
  //     method: false ? 'PUT' : 'POST',
  //     headers: { 'Content-Type': 'application/json' },
  //     body: data,
  //   })
  //     .then(res => res.json())
  //     .then((response) => {
  //       dispatch(showMoviePopup(false));
  //       if (!isEditMode) { dispatch(addMovieToList(response)); }
  //       return history.push('/');
  //     })
  //     .catch(() => console.log('error'));
  // }, [history, dispatch]);

  const onReset = () => {
    form.resetFields();
  };

  const handleSubmit = (values) => {
    values.id = getMovieId();
    values.date = values.date.format('YYYY/MM/DD');;
    // addMovieToDb(JSON.stringify(values));
  }

  const getMovieId = () => (Math.random().toString(36).replace('0.', ''));

  return (
    <Modal
      visible={moviePopupShown}
      title='Add Movie'
      onCancel={closePopup}
      footer={null}>
      <Form className="moviePopup" {...layout} form={form} onFinish={handleSubmit}>
        <Form.Item name="title" label="Title"
          rules={[{ required: true, },]}>
          <Input />
        </Form.Item>
        <Form.Item name="type" label="Type"
          rules={[{ required: true, },]}>
          <Select allowClear>
            <Option value="Movie">Movie</Option>
            <Option value="Series">Series</Option>
            <Option value="Documentary">Documentary</Option>
          </Select>
        </Form.Item>
        <Form.Item name="rating" label="Rating"
          rules={[{ required: true, },]}>
          <Select allowClear>
            <Option value="1">1</Option>
            <Option value="2">2</Option>
            <Option value="3">3</Option>
            <Option value="4">4</Option>
            <Option value="5">5</Option>
          </Select>
        </Form.Item>
        <Form.Item name="date" label="Released on"
          rules={[{ required: true, },]}>
          <DatePicker />
        </Form.Item>
        <Form.Item name="poster" label="Poster"
          rules={[{ required: true, },]}>
          <Input />
        </Form.Item>
        <Form.Item name="description" label="Description"
          rules={[{ required: true, },]}>
          <TextArea placeholder='Description...' autoSize />
        </Form.Item>
        <Form.Item {...tailLayout}>
          <Button htmlType="button" onClick={onReset}>Reset</Button>
          <Button type="primary" htmlType="submit">Submit</Button>
        </Form.Item>
      </Form>
    </Modal>
  )
}

export default MoviePopup;