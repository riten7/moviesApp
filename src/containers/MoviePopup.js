import React from 'react';
import { Modal, Form, Input, Button, Select, DatePicker, Spin } from 'antd';
import moment from 'moment';

const layout = { labelCol: { span: 6 }, wrapperCol: {span: 16 },};
const tailLayout = { wrapperCol: { offset: 16 }};

const MoviePopup = (props) => {
  const [form] = Form.useForm();
  const { Option } = Select;
  const { TextArea } = Input;
  
  const closePopup = () => props.setPopupState(false);

  const getInitialValues = () => {
    if (props.movie) {
      return {
        title: props.movie.title,
        type: props.movie.type,
        rating: parseInt(props.movie.rating),
        poster: props.movie.poster,
        description: props.movie.description,
        date: moment(props.movie.date, 'YYYY/MM/DD')
      }
    }
  }

  const onReset = () => {
    form.resetFields();
  };
  
  const handleSubmit = (values) => {
    values.date = values.date.format('YYYY/MM/DD');
    props.submitMovie(values);
    closePopup();
  };

  return (
    <>
    <Modal
      visible={true}
      title= {props.movie ? 'Edit Movie' : 'Add Movie'}
      onCancel={closePopup}
      footer={null}
      mask={true}
      maskClosable={false}>
      <Form className="moviePopup" {...layout} form={form} initialValues={getInitialValues()} onFinish={handleSubmit}>
        <Form.Item name="title" label="Title"
          rules={[{ required: true }]}>
          <Input />
        </Form.Item>
        <Form.Item name="type" label="Type"
        rules={[{ required: true }]}>
          <Select allowClear>
            <Option value="Movie">Movie</Option>
            <Option value="Series">Series</Option>
            <Option value="Documentary">Documentary</Option>
          </Select>
        </Form.Item>
        <Form.Item name="rating" label="Rating"
          rules={[{ required: true }]}>
          <Select allowClear>
            <Option value="1">1</Option>
            <Option value="2">2</Option>
            <Option value="3">3</Option>
            <Option value="4">4</Option>
            <Option value="5">5</Option>
          </Select>
        </Form.Item>
        <Form.Item name="date" label="Released On"
          rules={[{ required: true }]}>
          <DatePicker />
        </Form.Item>
        <Form.Item name="poster" label="Poster"
          rules={[{ required: true }]}>
          <Input />
        </Form.Item>
        <Form.Item name="description" label="Description"
          rules={[{ required: true }]}>
          <TextArea placeholder='Description...' autoSize />
        </Form.Item>
        <Form.Item {...tailLayout}>
          <Button htmlType="button" onClick={onReset}>Reset</Button>
          <Button type="primary" htmlType="submit">Submit</Button>
        </Form.Item>
      </Form>
    </Modal>
    </>
  )
}

export default MoviePopup;