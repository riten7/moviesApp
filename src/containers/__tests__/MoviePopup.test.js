import React from "react";
import { shallow, mount } from "enzyme";
import MoviePopup from "../MoviePopup";
import configureStore from "redux-mock-store";
import thunk from 'redux-thunk';
import * as ReactReduxHooks from "../react-redux-hooks";
import {Provider} from 'react-redux';
import data from '../__mockData__/movielistData.json';
import moment from 'moment';

const ipdata = {
      title: 'ytr',
      type: 'hj',
      rating: '1',
      poster: 'hgfgh',
      description: 'jkjlk',
      date: moment('2020/01/01', 'YYYY/MM/DD')}

window.matchMedia = jest.fn().mockImplementation((query) => ({
  matches: false,
  media: query,
  onchange: null,
  addListener: jest.fn(),
  removeListener: jest.fn(),
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  dispatchEvent: jest.fn(),
}))

describe("Component renders properly", () => {
  let store;
  let wrapper;
   beforeEach(() => {
    store = configureStore([thunk])(data);
    jest
      .spyOn(ReactReduxHooks, "useSelector")
      .mockImplementation(state => store.getState());
    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);
  });

  it("Test Submit function works", () => {
    const setPopupState= jest.fn();
    const submitMovie = jest.fn();
    wrapper = mount(<Provider store={store}><MoviePopup 
    setPopupState={setPopupState}
    submitMovie={submitMovie}/></Provider>);
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.find('.moviePopup').at(0).props().form.setFieldsValue(ipdata);
    wrapper.find('.ant-btn.ant-btn-primary').simulate('submit');
    //expect(submitMovie).toHaveBeenCalled();
  });
});