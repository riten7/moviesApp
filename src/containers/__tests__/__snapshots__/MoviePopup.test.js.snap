// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders properly Test Submit function works 1`] = `"<div class=\\"ant-modal-root\\"><div class=\\"ant-modal-mask fade-appear\\"></div><div tabindex=\\"-1\\" class=\\"ant-modal-wrap \\" role=\\"dialog\\" aria-labelledby=\\"rcDialogTitle0\\"><div role=\\"document\\" style=\\"width: 520px;\\" class=\\"ant-modal zoom-appear\\"><div tabindex=\\"0\\" style=\\"width: 0px; height: 0px; overflow: hidden; outline: none;\\" aria-hidden=\\"true\\"></div><div class=\\"ant-modal-content\\"><button type=\\"button\\" aria-label=\\"Close\\" class=\\"ant-modal-close\\"><span class=\\"ant-modal-close-x\\"><span role=\\"img\\" aria-label=\\"close\\" class=\\"anticon anticon-close ant-modal-close-icon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"close\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z\\"></path></svg></span></span></button><div class=\\"ant-modal-header\\"><div class=\\"ant-modal-title\\" id=\\"rcDialogTitle0\\">Add Movie</div></div><div class=\\"ant-modal-body\\"><form class=\\"ant-form ant-form-horizontal moviePopup\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-6 ant-form-item-label\\"><label for=\\"title\\" class=\\"ant-form-item-required\\" title=\\"Title\\">Title</label></div><div class=\\"ant-col ant-col-16 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><input type=\\"text\\" id=\\"title\\" class=\\"ant-input\\" value=\\"\\"></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-6 ant-form-item-label\\"><label for=\\"type\\" class=\\"ant-form-item-required\\" title=\\"Type\\">Type</label></div><div class=\\"ant-col ant-col-16 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow\\"><div class=\\"ant-select-selector\\"><span class=\\"ant-select-selection-search\\"><input id=\\"type\\" autocomplete=\\"off\\" class=\\"ant-select-selection-search-input\\" style=\\"opacity: 0;\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"type_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"type_list\\" aria-activedescendant=\\"type_list_0\\" readonly=\\"\\" value=\\"\\"></span><span class=\\"ant-select-selection-placeholder\\"></span></div><span class=\\"ant-select-arrow\\" style=\\"user-select: none;\\" unselectable=\\"on\\" aria-hidden=\\"true\\"><span role=\\"img\\" aria-label=\\"down\\" class=\\"anticon anticon-down\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z\\"></path></svg></span></span></div></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-6 ant-form-item-label\\"><label for=\\"rating\\" class=\\"ant-form-item-required\\" title=\\"Rating\\">Rating</label></div><div class=\\"ant-col ant-col-16 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow\\"><div class=\\"ant-select-selector\\"><span class=\\"ant-select-selection-search\\"><input id=\\"rating\\" autocomplete=\\"off\\" class=\\"ant-select-selection-search-input\\" style=\\"opacity: 0;\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"rating_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"rating_list\\" aria-activedescendant=\\"rating_list_0\\" readonly=\\"\\" value=\\"\\"></span><span class=\\"ant-select-selection-placeholder\\"></span></div><span class=\\"ant-select-arrow\\" style=\\"user-select: none;\\" unselectable=\\"on\\" aria-hidden=\\"true\\"><span role=\\"img\\" aria-label=\\"down\\" class=\\"anticon anticon-down\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z\\"></path></svg></span></span></div></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-6 ant-form-item-label\\"><label for=\\"date\\" class=\\"ant-form-item-required\\" title=\\"Released On\\">Released On</label></div><div class=\\"ant-col ant-col-16 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-picker\\"><div class=\\"ant-picker-input\\"><input readonly=\\"\\" placeholder=\\"Select date\\" title=\\"\\" size=\\"12\\" value=\\"\\"><span class=\\"ant-picker-suffix\\"><span role=\\"img\\" aria-label=\\"calendar\\" class=\\"anticon anticon-calendar\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"calendar\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z\\"></path></svg></span></span></div></div></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-6 ant-form-item-label\\"><label for=\\"poster\\" class=\\"ant-form-item-required\\" title=\\"Poster\\">Poster</label></div><div class=\\"ant-col ant-col-16 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><input type=\\"text\\" id=\\"poster\\" class=\\"ant-input\\" value=\\"\\"></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-6 ant-form-item-label\\"><label for=\\"description\\" class=\\"ant-form-item-required\\" title=\\"Description\\">Description</label></div><div class=\\"ant-col ant-col-16 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><textarea placeholder=\\"Description...\\" id=\\"description\\" class=\\"ant-input\\" style=\\"height: 0px; min-height: -9007199254740991px; max-height: 9007199254740991px; overflow-y: hidden; overflow-x: hidden;\\"></textarea></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-col-offset-16 ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><button type=\\"button\\" class=\\"ant-btn\\"><span>Reset</span></button><button type=\\"submit\\" class=\\"ant-btn ant-btn-primary\\"><span>Submit</span></button></div></div></div></div></form></div></div><div tabindex=\\"0\\" style=\\"width: 0px; height: 0px; overflow: hidden; outline: none;\\" aria-hidden=\\"true\\"></div></div></div></div>"`;
